### PR TITLE
Use WS for operator load/save

### PR DESCRIPTION
**Pull request by brian-brazil**
Trim down options to what actually works. There's
no distinction between everything and scoreboard now,
and Pages hasn't existed in a while.

Users should use Start New Game, not resetting manually,
so the reset can be removed.

The full reset doesn't work, as it depends on the scoreboard correctly
resetting itself - which it doesn't completely and in the event of a bug
isn't likely to help you anyway if you've a bug like that (or tried to
merge/load in something that you shouldn't have). Blowing away the save
and restarting will always work though.

The new way of doing things seems a bit more fragile than the old pure-XML way, I wonder if we should restrict loading to PreparedTeams.

### CODE REVIEW

File: src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java [hunk 1]
```diff
@@ -51,34 +52,53 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
     }
 
-    protected void handleDocument(HttpServletRequest request, HttpServletResponse response, Document doc) throws IOException {
+    protected void handleJSON(HttpServletRequest request, HttpServletResponse response, JSONObject json) throws IOException {
+        List<ScoreBoardJSONSetter.JSONSet> jsl = new ArrayList<ScoreBoardJSONSetter.JSONSet>();
+
+        JSONObject state = json.getJSONObject("state");
+        Flag flag = null;
+        if (request.getPathInfo().equalsIgnoreCase("/load")) {
+            flag = Flag.FROM_AUTOSAVE;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java]: This should be set for merge as well. (The fact that it isn't set for merging XML files is a bug that .)

File: src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java [hunk 2]
```diff
@@ -51,34 +52,53 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
             response.sendError(HttpServletResponse.SC_BAD_REQUEST, fuE.getMessage());
-        } catch ( JDOMException jE ) {
+        } catch ( JSONException jE ) {
             response.sendError(HttpServletResponse.SC_BAD_REQUEST, jE.getMessage());
         }
     }
 
-    protected void handleDocument(HttpServletRequest request, HttpServletResponse response, Document doc) throws IOException {
+    protected void handleJSON(HttpServletRequest request, HttpServletResponse response, JSONObject json) throws IOException {
+        List<ScoreBoardJSONSetter.JSONSet> jsl = new ArrayList<ScoreBoardJSONSetter.JSONSet>();
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java]: The type argument on ArrayList is redundant.

File: src/com/carolinarollergirls/scoreboard/jetty/WS.java [hunk 3]
```diff
@@ -44,8 +43,9 @@
 import com.carolinarollergirls.scoreboard.event.ScoreBoardEventProvider;
 import com.carolinarollergirls.scoreboard.event.ScoreBoardEventProvider.Flag;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/WS.java:39]: All includes from .event except .Flag are unused.

File: src/com/carolinarollergirls/scoreboard/jetty/WS.java [hunk 4]
```diff
@@ -44,8 +43,9 @@
...
 import com.carolinarollergirls.scoreboard.json.JSONStateManager;
-import com.carolinarollergirls.scoreboard.utils.PropertyConversion;
 import com.carolinarollergirls.scoreboard.json.JSONStateListener;
+import com.carolinarollergirls.scoreboard.json.ScoreBoardJSONSetter;
+import com.carolinarollergirls.scoreboard.utils.PropertyConversion;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/WS.java]: This is unused.

File: src/com/carolinarollergirls/scoreboard/json/ScoreBoardJSONSetter.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.regex.Pattern;
+
+import com.carolinarollergirls.scoreboard.ScoreBoardManager;
+import com.carolinarollergirls.scoreboard.core.ScoreBoard;
+import com.carolinarollergirls.scoreboard.event.OrderedScoreBoardEventProvider.IValue;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.AddRemoveProperty;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.CommandProperty;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.PermanentProperty;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.Property;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.ValueWithId;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/json/ScoreBoardJSONSetter.java:23]: This is unused.

File: html/controls/operator.js [hunk 6]
```diff
@@ -2209,21 +2209,19 @@ function createSaveLoadContent() {
...
-		.append("<td colspan='4'>To download, right-click and Save - to view XML, left-click</td>");
+		.append("<td colspan='4'>To download, right-click and Save - to view JSON, left-click</td>");
 	var contentRow = $("<tr>").addClass("Content").appendTo(sbDownloadTable);
 
 	var links = [
 		{ name: "All data", url: "" },
-		{ name: "ScoreBoard", url: "scoreboard.xml?path=ScoreBoard" },
-		{ name: "Teams", url: "teams.xml?path=Teams" },
-		{ name: "Pages", url: "pages.xml?path=Pages" }
+		{ name: "Teams", url: "teams.json?path=ScoreBoard.PreparedTeam" }
```
Comment (frank-weinberg) [inline html/controls/operator.js:2219]: Should we add the game data file for the current game to this menu?

File: src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java (hunk 1 above)
Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java]: Done

File: src/com/carolinarollergirls/scoreboard/jetty/WS.java (hunk 3 above)
Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/jetty/WS.java:39]: Removed

File: src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java (hunk 2 above)
Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/jetty/LoadJsonScoreBoard.java]: Done

File: html/controls/operator.js (hunk 6 above)
Comment (brian-brazil) [reply html/controls/operator.js:2219]: Not a bad idea, but we'd probably want to re-think what counts as game-data first. The full scoreboard can serve the same purpose for now.

File: src/com/carolinarollergirls/scoreboard/jetty/SaveJsonScoreBoard.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+public class SaveJsonScoreBoard extends HttpServlet {
+
+    public SaveJsonScoreBoard(JSONStateManager jsm) {
+      this.jsm = jsm;
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException,IOException {
+        // Use  a TreeMap to keep output sorted.
+        Map<String, Object> state = new TreeMap<String, Object>(jsm.getState());
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/jetty/SaveJsonScoreBoard.java]: Type Arguments on TreeMap are also redundant.

Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/jetty/SaveJsonScoreBoard.java]: Done

**Pull request merged**